astropy/modeling/fitting.py

```diff
@@ -763,6 +764,13 @@ def __call__(self, model, x, y, z=None, weights=None, **kwargs):
                                            filtered_data,
                                            weights=filtered_weights, **kwargs)
 
+            # Stop iteration if the masked points are no longer changing (with
+            # cumulative rejection we only need to compare how many there are):
+            this_n_masked = filtered_data.mask.sum()  # (minimal overhead)
+            if this_n_masked == last_n_masked:
+                break
+            last_n_masked = this_n_masked
```

Comment:
I considered skipping this whole block when the loop has reached `niter` anyway, mostly to avoid the unnecessary `sum()`, but the difference in performance seems to be in the noise for a small (100 element) array, so I think it's not worth complicating the code slightly for this.